osu.Framework/Graphics/Containers/GridContainer.cs

```diff
@@ -37,20 +37,31 @@ private void load()
         /// Null elements are allowed to represent blank rows/cells.
         /// </para>
         /// </summary>
-        public Drawable[][] Content
+        public GridContainerContent Content
         {
             get => content;
             set
             {
-                if (content == value)
+                if (content?.Equals(value) ?? false)
```

Comment:
```suggestion
                if (content?.Equals(value) == true)
```

reads better imho and is how we do this elsewhere.